### DISCUSSION TITLE
How to disable LSP (bashls) on a per-file basis?

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [LazyVim/LazyVim#2347] [Q&A] [answered]
I noticed that the `-- stylua: ignore` is used for `stylua`, but it appears that this approach doesn't work as expected for bashls.

**Comment by dpetka2001** [comment, marked as answer]
`bashls` is a language server. `stylua` is used for formatting Lua files. 

`-- stylua: ignore` is used for defining specific scopes in a lua file where the formatter shouldn't take the specific scope into consideration, thus leaving it be as the user writes it.

What exactly do you want to achieve? I found in a [comment](https://github.com/bash-lsp/bash-language-server/issues/738#issuecomment-1454704515) in `bashls` git repo that it supports parsing `shellcheck` directives. Look [here](https://github.com/koalaman/shellcheck/wiki/Ignore) and [here](https://github.com/koalaman/shellcheck/wiki/Directive) for how to disable specific errors you might not want to see. 

I don't use it myself, so you should read the above resources to configure it how you want it to behave.

**Reply by limaceous-bushwhacker** [reply]
Tysm, this is great. Had no idea we could modify ShellCheck errors. This does what I'm looking for: https://github.com/koalaman/shellcheck/wiki/Ignore#ignoring-all-errors-in-a-file-08
